### PR TITLE
Feature/basic interface for the forum (v2)

### LINKED ISSUE #1159: Basic interface for the forum
## Overview

The forum is a feature which is already implemented and deployed on "dev" deployment. For instance,
- In a first tab, as *demo user*, navigate to [this task](https://dev.algorea.org/activities/by-id/6379723280369399253;path=4702,1625159049301502151,5207993233955027100;attempId=0/details)
- In a second tab, * as the usual user *, navigate to [this task](https://dev.algorea.org/activities/by-id/6379723280369399253;path=4702,1625159049301502151,5207993233955027100;attempId=0/details?watchedGroupId=752024252804317630&watchUser=1) 

## Acceptance criteria
- [ ] ...

### PR DESCRIPTION
**Pull request by smadbe**
Copy/rewrite from #1175.

## Description

Fixes #1159 

## Test cases

- [ ] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/floating-discussions/en/activities/by-id/7197346214473712928;path=4702,1625159049301502151,5207993233955027100;attempId=0/details)
  3. And I click on open thread button in top right corner
  4. Then I see opened thread panel right of screen

- [ ] Case 2:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/floating-discussions/en/activities/by-id/7197346214473712928;path=4702,1625159049301502151,5207993233955027100;attempId=0/details)
  3. And I click on open full left menu  button
  4. Then I click on open thread button on right side
  5. Then I see opened thread panel right of screen

- [ ] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/floating-discussions/en/activities/by-id/7197346214473712928;path=4702,1625159049301502151,5207993233955027100;attempId=0/details)
  3. And I click on open thread button in top right corner
  4. Then I see opened thread panel right of screen
  5. And I click on open full left menu  button
  6. Then I go to "Test task to delete afterwards"
  7. Then I see the thread was disappeared and button thread is unselected

- [ ] Case 4:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/floating-discussions/en/activities/by-id/7197346214473712928;path=4702,1625159049301502151,5207993233955027100;attempId=0/details)
  3. And I click on open thread button in top right corner
  4. Then I see opened thread panel right of screen
  5. And I send new message
  6. Then I see new message in list of messages

- [ ] Case 5:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/floating-discussions/en/activities/by-id/7197346214473712928;path=4702,1625159049301502151,5207993233955027100;attempId=0/details)
  3. And I click on open thread button in top right corner
  4. Then I see opened thread panel right of screen
  5. And I solve Task version 1*
  6. Then I see new message with score

### CODE REVIEW

**Comment by smadbe** [comment]
So this is the rewrite of #1175 which uses a new global service "DiscussionService" to control if the thread has to be displayed and how.

@Iloveall , I've included your css and buttons from #1175 but I haven't touched the design of the thread component except removing the show/hide button.  Could you re-apply your changes ? 

Probably you should now create a "ThreadContainerComponent" and make sure all animation stuffs goes in it not to pollute the item-details one. If it is easier for you to make it "float" on top for now, make it float.

Details: could the icon be a bit larger in the button for opening/closing the panel?

Let's not use the "unread count" for now, it will be for a future PR.

File: src/app/modules/item/components/thread-container/thread-container.component.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.__container {
```
Comment (smadbe) [inline src/app/modules/item/components/thread-container/thread-container.component.scss:1]: what's the semantic of the double underscore?

**Review by smadbe — COMMENTED** [review]
Looks good 👍

File: src/app/modules/item/components/thread-container/thread-container.component.scss (hunk 1 above)
Comment (Iloveall) [reply src/app/modules/item/components/thread-container/thread-container.component.scss]: This is BEM methodology, my old habit, so i've added it by mistake. Hope in future we'll use some methodology as well.

Fixed

**Review by Iloveall — APPROVED** [review]

**Pull request merged**